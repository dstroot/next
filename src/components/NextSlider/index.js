import React, { useState } from 'react';
import './styles.scss';

// slider component
// https://github.com/react-component/slider/
import Slider from 'rc-slider/lib/Slider';
import 'rc-slider/assets/index.css';

const NextSlider = () => {
  const [value, setValue] = useState(500);

  // set value to slider position
  const handleChange = num => {
    setValue(num);
  };

  return (
    <div className="container">
      <div className="row">
        <div className="col-md-6">
          <div className="mt-4">
            <div className="d-flex justify-content-between">
              <span>Intial Value</span>
              <span>{value}</span>
            </div>
            <Slider
              min={1}
              max={1000}
              defaultValue={value}
              onChange={handleChange}
            />
          </div>
        </div>
      </div>
    </div>
  );
};

export default NextSlider;
