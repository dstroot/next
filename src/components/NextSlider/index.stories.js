import React from 'react';
import { storiesOf } from '@storybook/react';
import NextSlider from '../NextSlider';

storiesOf('NextSlider', module)
  .addParameters({
    info: {
      inline: true,
      header: false,
      text: `This supports markdown!

      ~~~js
      console.log("hello");
      ~~~
      `,
    },
  })
  .add('show NextSlider', () => <NextSlider />);
