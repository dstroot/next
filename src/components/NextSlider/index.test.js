import React from 'react';
import { create } from 'react-test-renderer';
import { mount } from 'enzyme';
import NextSlider from '../NextSlider';

describe('NextSlider', () => {
  // mount the component
  let mountedComponent;
  const getMountedComponent = () => {
    if (!mountedComponent) {
      mountedComponent = mount(<NextSlider />);
    }
    return mountedComponent;
  };

  beforeEach(() => {
    mountedComponent = undefined;
  });

  it('it should render', () => {
    let tree = create(<NextSlider />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
