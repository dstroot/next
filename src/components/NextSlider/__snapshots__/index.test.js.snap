// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NextSlider it should render 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-md-6"
    >
      <div
        className="mt-4"
      >
        <div
          className="d-flex justify-content-between"
        >
          <span>
            Intial Value
          </span>
          <span>
            500
          </span>
        </div>
        <div
          className="rc-slider"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          onTouchStart={[Function]}
        >
          <div
            className="rc-slider-rail"
            style={Object {}}
          />
          <div
            className="rc-slider-track"
            style={
              Object {
                "left": "0%",
                "width": "49.94994994994995%",
              }
            }
          />
          <div
            className="rc-slider-step"
          />
          <div
            aria-disabled={false}
            aria-valuemax={1000}
            aria-valuemin={1}
            aria-valuenow={500}
            className="rc-slider-handle"
            onBlur={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="slider"
            style={
              Object {
                "left": "49.94994994994995%",
              }
            }
            tabIndex={0}
          />
          <div
            className="rc-slider-mark"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
